src/apps/hw.js

```diff
@@ -70,20 +74,29 @@ export const streamAppInstall = (
         }
         if (e.type === "result") {
           // stream install with the result of list apps
-          const state = reducer(initState(e.result), {
-            type: "install",
-            name: appName,
-          });
-          if (isOutOfMemoryState(predictOptimisticState(state))) {
+          let state = initState(e.result);
+          appNames.forEach(
+            (name) => (state = reducer(state, { type: "install", name }))
+          );
+
+          if (
+            isOutOfMemoryState(predictOptimisticState(state)) ||
+            !getEnv("EXPERIMENTAL_INLINE_INSTALL")
```

Comment:
i don't understand why the `if` is here.

Reply:
Doing the listapps and install plan allows me to know what _would be_ installed even if I don't finally do it because the user didn't opt in for inline installation. I need a way of knowing what is the error case data if you get what I mean